### PR TITLE
Add from_geoh5 for Points and data. Unittest

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-637 - convert geoh5 Points to OMF Pointsets**

### CODE REVIEW

File: omf/fileio/geoh5.py [hunk 1]
```diff
@@ -170,9 +208,22 @@ def from_omf(self, parent=None, **kwargs) -> Data:
...
         return self._entity
 
-    def from_geoh5(self) -> UidModel:
-        """TODO Convert geoh5 entity to omf element."""
-        raise NotImplementedError
+    def from_geoh5(self, **kwargs) -> UidModel:
+        """Convert a geoh5 data to omf element."""
+        with fetch_h5_handle(self.geoh5):
+            kwargs = self.collect_h5_attributes(**kwargs)
+            uid = kwargs.get("uid")
```
Comment (benk-mira) [inline omf/fileio/geoh5.py:215]: Just use pop instead of get and you can drop the del statement!

File: omf/fileio/geoh5.py [hunk 2]
```diff
@@ -181,52 +232,73 @@ class ElementConversion(BaseConversion):
...
         return self._entity
 
-    def process_dependents(self, workspace):
-        if getattr(self.element, "data", None):
-            for child in self.element.data:
+    def from_geoh5(self, **kwargs) -> UidModel:
+        """Convert a geoh5 entity to omf element."""
+        with fetch_h5_handle(self.geoh5) as workspace:
+            kwargs = self.collect_h5_attributes(**kwargs)
+            uid = kwargs.get("uid")
```
Comment (benk-mira) [inline omf/fileio/geoh5.py:267]: It's like I _always_ say Dom:

"When the pigs try and get at you..."

"`pop` it like it's Haaawt, `pop` it like it's` haaawt."

File: omf/fileio/geoh5.py [hunk 3]
```diff
@@ -181,52 +232,73 @@ class ElementConversion(BaseConversion):
...
+            for child in parent.data:
                 converter = _CONVERSION_MAP[type(child)](child, workspace)
                 converter.from_omf(parent=self.entity)
 
-    def from_geoh5(self) -> UidModel:
-        """TODO Convert geoh5 entity to omf element."""
-        raise NotImplementedError
+        elif isinstance(parent, Entity) and getattr(parent, "children", None):
+            data = []
+            for child in parent.children:
```
Comment (benk-mira) [inline omf/fileio/geoh5.py:283]: Fails if parent.data is None

File: omf/fileio/geoh5.py [hunk 4]
```diff
@@ -181,52 +232,73 @@ class ElementConversion(BaseConversion):
...
+            del kwargs["uid"]
+            self._element = self.omf_type(**kwargs)
+            self._element._backend.update({"uid": uid})  # pylint: disable=W0212
+            self.process_dependents(self.entity, workspace)
+
+        return self._element
+
+    def process_dependents(self, parent: UidModel | Entity, workspace):
+        if isinstance(parent, UidModel) and getattr(parent, "data", None):
+            for child in parent.data:
```
Comment (benk-mira) [inline omf/fileio/geoh5.py:277]: Fails if parent.data is None

**Review by benk-mira — CHANGES_REQUESTED** [review]
Some small feedback.  Don't miss - hilarious joke contained inside!

File: omf/fileio/geoh5.py (hunk 1 above)
Comment (domfournier) [reply omf/fileio/geoh5.py:215]: Nice, of course. Forgot about that one.

File: omf/fileio/geoh5.py (hunk 2 above)
Comment (domfournier) [reply omf/fileio/geoh5.py:267]: ahaha lol. Somehow I pictured John Oliver making that joke

**Pull request merged**